### PR TITLE
Normalize page title slug after access check

### LINKED ISSUE #6416: Custom menu pages title leaked when access is deined
**Describe the bug**
The custom pages you can add to event menus all have an id. You can cycle through them by visiting e.g. https://events.example.com/event/12/page/345 which will redirect to https://events.example.com/event/12/page/345-sekrit-page . If the user doesn't have permission for that page, they can still read the title via the URL slug.

**To Reproduce**
1. Create custom pages that have access restrictions
2. Visit the URL in a private window
3. See redirected URL title

**Expected behavior**
Access restrictions should be applied before the redirect occurs. If the user has no access, the URL should not redirect.

### PR DESCRIPTION
**Pull request by ThiefMaster**
closes #6416

### CODE REVIEW

File: indico/web/rh.py [hunk 1]
```diff
@@ -77,6 +77,9 @@ class RH:
         'endpoint': None
     }
 
+    #: Like `normalize_url_spec`, but runs after the access check succeeded
+    normalize_url_spec_late = None
```
Comment (tomasr8) [inline indico/web/rh.py:81]: Not a very descriptive name, but anything else I can think of is too wordy :/

Comment (ThiefMaster) [reply indico/web/rh.py:81]: yeah, couldn't think of anything nicer either...
